src/pages/user/components/ReportModal.tsx

```diff
@@ -0,0 +1,139 @@
+
+export const ReportModal = () => {
+  const { loggedInUser } = useAuthContext()
+  const [largeScreen] = useMediaQuery('(min-width: 48em)')
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const [startDate, setStartDate] = useState(getStartOfSemester())
+  const [endDate, setEndDate] = useState(getEndOfSemester())
+  const [adminReport, setAdminReport] = useState(false)
+
+  const getUserReportMutation = useUserReportMutation()
+  const userReportRef = useRef<HTMLButtonElement>(null)
+  const getAdminReportMutation = useAdminReportMutation()
+  const adminReportRef = useRef<HTMLButtonElement>(null)
+
+  const startDateError = startDate >= endDate
+  const endDateError = endDate > new Date()
+
+  const onEndDateChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const date = new Date(e.target.value)
+    date.setHours(23, 59, 59, 999)
+    setEndDate(date)
+  }
+
+  return (
+    <>
+      <Button colorScheme="green" onClick={onOpen}>
+        Riport generálása
+      </Button>
+      <Modal isCentered={largeScreen} motionPreset="slideInBottom" isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Riport generálása</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody pb={6}>
+            <VStack alignItems="flex-start">
+              <Text>HK-s ösztöndíjakhoz itt tudsz riport generálni az általad tartott konzultációkról.</Text>
+              {loggedInUser?.isAdmin && (
+                <Text textAlign="justify">
+                  Adminként van lehetőséged olyan riport generálására, melyen az összes konzi szereplni fog, amit az adott időtartamban
+                  tartottak.
+                </Text>
+              )}
+              <FormControl>
+                <FormLabel>Időtartam kezdete</FormLabel>
+                <Input
+                  type="date"
+                  onChange={(e) => setStartDate(new Date(e.target.value))}
+                  value={formatDate(startDate)}
+                  max={formatDate(new Date(Date.now() - 24 * 60 * 60 * 1000))}
+                />
+
+                <Flex justifyContent="flex-end">
+                  {startDateError && <FormHelperText color="red.500">Érvénytelen időtartam!</FormHelperText>}
+                </Flex>
+              </FormControl>
+              <FormControl>
+                <FormLabel>Időtartam vége</FormLabel>
+                <Input type="date" onChange={onEndDateChange} value={formatDate(endDate)} max={formatDate(new Date())} />
+
+                <Flex justifyContent="flex-end">
+                  {endDateError && <FormHelperText color="red.500">Csak múltbeli konzikról lehet riportot generálni!</FormHelperText>}
+                </Flex>
+              </FormControl>
+              {loggedInUser?.isAdmin && (
+                <Checkbox colorScheme="green" isChecked={adminReport} onChange={(e) => setAdminReport(e.target.checked)}>
+                  Admin riport
+                </Checkbox>
```

Comment:
Reset the dates and the checkbox on close

Reply:
This was intentional. If the user reopens the modal, they were probably dissatisfied with the report, so they want a new one, likely one that is similiar to the previous, so I think it helps if we don't reset the inputs.

However I did forget to close the modal when the report is done